### DISCUSSION TITLE
Need to implement a null mediator pattern

### DISCUSSION THREAD

**Discussion by brunobozic** [LuckyPennySoftware/MediatR#865] [Q&A] [answered]
Need a null mediatr along the lines of following:

```csharp
public class NoMediator : IMediator
        {

            public  IAsyncEnumerable<TResponse> CreateStream<TResponse>(IStreamRequest<TResponse> request, CancellationToken cancellationToken = default)
            {
                return default(IAsyncEnumerable<TResponse>);
            }

            public  IAsyncEnumerable<object> CreateStream(object request, CancellationToken cancellationToken = default)
            {
                return default(IAsyncEnumerable<object>);
            }

            public  Task Publish(object notification, CancellationToken cancellationToken = default)
            {
                return Task.CompletedTask;
            }

            public  Task Publish<TNotification>(TNotification notification, CancellationToken cancellationToken = default) where TNotification : INotification
            {
                return Task.CompletedTask;
            }

            public async Task<TResponse> Send<TResponse>(IRequest<TResponse> request, CancellationToken cancellationToken = default)
            {
                return default(TResponse);
            }

            public async Task<object> Send(object request, CancellationToken cancellationToken = default)
            {
                return Task.CompletedTask;
            }
        }
```

The reason for this is - I supply it into DbContext because I need mediatR to dispatch uncommited domain events from there.
I also need to use a Factory (for migrations and stuff) that shares the same interface and hence requires the mediatR depentency to be satisfied. 

What I have been getting since taking the latest version is:

`Method 'Send' in type 'NoMediator' from assembly 'Assembly.EFCore, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.'`

Any ideas?

**Comment by jbogard** [comment, marked as answer]
It looks like you need to update your implementation. There were new methods added to `IMediator`.

**Reply by brunobozic** [reply]
You are right, I had two different versions (of MediatR) in two different csproj under my sln, and the container targeted the older one, that one had an older signature for the "Send" method and this got me confused.

Thanks for the answer though, it immediately led me to check the versions of MediatR that I use and this led to the solution.
